### PR TITLE
Merging latest QIR work, including LLVMSharp dependency, into feature branch

### PR DESCRIPTION
**Pull request by swernli**
The merges has the latest bug fixes and updates for QIR generation, including the addition of stand-alone LLVM generation via a hybrid of Llvm.NET and LLVMSharp, eliminating the need for any cached binaries while still support cross-platform development.

### CODE REVIEW

File: .fantomasignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+﻿# Ignore signature files because of formatting bugs.
```
[the remark this answers is not in the document]
Comment (swernli) [reply .fantomasignore:1]: Yes, this one comes from a change Sarah did, I believe... https://github.com/microsoft/qsharp-compiler/pull/845

File: examples/VQE/EstimateEnergy.qs [hunk 2]
```diff
@@ -0,0 +1,127 @@
+namespace Microsoft.Quantum.Samples.Chemistry.SimpleVQE.EstimateEnergy {
```
[the remark this answers is not in the document]
Comment (swernli) [reply examples/VQE/EstimateEnergy.qs:1]: Ah, got it. Let me pull those out, thanks for catching!

File: src/QsCompiler/Llvm.NET/LLVMSharpExtensions/LLVMMemoryBufferRef.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
```
[the remark this answers is not in the document]
Comment (swernli) [reply src/QsCompiler/Llvm.NET/LLVMSharpExtensions/LLVMMemoryBufferRef.cs:1]: Yes, that's correct. There are places where LLVMSharp was missing accessors for functionality that the bindings needed (for example, to be able to use the code in an iterator). I wrote each of these files to provide the extensions we needed to hook into the underlying functionality.

File: src/QsCompiler/QirGeneration/Llvm.Net/cgmanifest.json [hunk 4]
```diff
@@ -1,23 +0,0 @@
-{
```
[the remark this answers is not in the document]
Comment (swernli) [reply src/QsCompiler/QirGeneration/Llvm.Net/cgmanifest.json:1]: Good call! I've added the cgmanifest.json, pointing to the commit hash that this code was forked from.

File: QsCompiler.sln [hunk 5]
```diff
@@ -53,6 +53,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "DocumentationGenerator", "s
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "QirGeneration", "src\QsCompiler\QirGeneration\QirGeneration.csproj", "{7FF41827-C922-448A-A241-D3C28ADD14E1}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Llvm.NET", "src\QsCompiler\Llvm.NET\Llvm.NET.csproj", "{36FB68CC-533A-40D4-A2E9-597A35C6912C}"
```
[the remark this answers is not in the document]
Comment (swernli) [reply QsCompiler.sln:56]: I do like that name better for our usage. I'll update it here and throughout the code.

**Review by bettinaheim — APPROVED** [review]

**Pull request merged**